src/System.Device.Gpio/System/Device/Gpio/BoardIdentification .cs

```diff
@@ -0,0 +1,337 @@
+                    return Model.RaspberryPiZeroW;
+
+                case 0x2082:
+                    return Model.RaspberryPi3B;
+
+                case 0x20D3:
+                    return Model.RaspberryPi3BPlus;
+
+                case 0x20A0:
+                    return Model.RaspberryPiComputeModule3;
+
+                case 0x3111:
+                    return Model.RaspberryPi4;
+
+                default:
+                    return Model.Unknown;
+            }
+        }
+
+        /// <summary>
+        /// Gets the processor name.
+        /// </summary>
+        /// <value>
+        /// The name of the processor.
+        /// </value>
+        public string ProcessorName
+        {
+            get
+            {
+                _settings.TryGetValue("Hardware", out string hardware);
+                return hardware;
+            }
+        }
+
+        /// <summary>
+        /// Gets the board firmware version.
+        /// </summary>
+        public int Firmware
+        {
+            get
+            {
+                if (_settings.TryGetValue("Revision", out string revision)
+                    && !string.IsNullOrEmpty(revision)
+                    && int.TryParse(revision, NumberStyles.HexNumber, CultureInfo.InvariantCulture, out int firmware))
+                {
+                    return firmware;
+                }
+
+                return 0;
+            }
+        }
+
+        /// <summary>
+        /// Gets the serial number.
+        /// </summary>
+        public string SerialNumber
+        {
+            get
+            {
+                if (_settings.TryGetValue("Serial", out string serial)
+                    && !string.IsNullOrEmpty(serial))
+                {
+                    return serial;
+                }
+
+                return null;
+            }
+        }
+
+        /// <summary>
+        /// Gets a value indicating whether board is overclocked.
+        /// </summary>
+        /// <value>
+        ///   <c>true</c> if board is overclocked; otherwise, <c>false</c>.
+        /// </value>
+        public bool IsOverclocked
+        {
+            get
+            {
+                var firmware = Firmware;
+                return (firmware & 0xFFFF0000) != 0;
+            }
+        }
+
+        #endregion
+
+        #region Private Helpers
+
+        private Processor LoadProcessor(Model model)
+        {
+            switch (model)
+            {
+                case Model.RaspberryPiA:
+                case Model.RaspberryPiAPlus:
+                case Model.RaspberryPiBRev1:
+                case Model.RaspberryPiBRev2:
+                case Model.RaspberryPiBPlus:
+                case Model.RaspberryPiComputeModule:
+                case Model.RaspberryPiZero:
+                case Model.RaspberryPiZeroW:
+                    return Processor.Bcm2708;
+                case Model.RaspberryPi2B:
+                // TBC: B3(+) should be a BCM2710 processor ...
```

Comment:
I don't know, but the method is currently unused anyway. Not sure whether I should remove it for now, since it might still be useful later. Should probably rather use the Processor field now, since with the current kernel, that should work correctly.

Reply:
I'd vote to remove it if it's not used. You can create an issue which this would solve and paste the snippet so it doesn't get lost